subsys: net: download_client rework (#17481)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Cellular samples
----------------

> 

Reviewer: May be better to phrase the entries as:

```suggestion
* Updated the following samples to use the :ref:`lib_downloader` library instead of the deprecated :ref:`lib_download_client` library:
  
  * :ref:`http_application_update_sample`
  * :ref:`http_modem_delta_update_sample`
  * :ref:`http_modem_full_update_sample`
  * :ref:`lwm2m_carrier`
  * :ref:`lwm2m_client`
  * :ref:`modem_shell_application`
  * :ref:`nrf_cloud_multi_service`
  * :ref:`nrf_cloud_rest_fota`
```
The same could be followed for networking samples and networking libs.

Libraries for networking
------------------------

> * Added the :ref:`lib_downloader` library.
> 
> * :ref:`lib_download_client` library:
> 
>   * Deprecated the :ref:`lib_download_client` library. The application should instead use the :ref:`lib_downloader` library. See migration guide for changes required.

Reviewer: ```suggestion
* Added the :ref:`lib_downloader` library to be used instead of the deprecated :ref:`lib_download_client` library. 
  See the :ref:`migration guide <migration_2.9_recommended>` for required changes.
```
